core/tests/protractor.conf.js

```diff
@@ -158,7 +104,83 @@ exports.config = {
     users: [
       'protractor_desktop/userJourneys.js',
     ],
+  };
+
+// A reference configuration file.
+exports.config = {
+  // ----- How to setup Selenium -----
+  //
+  // There are three ways to specify how to use Selenium. Specify one of the
+  // following:
+  //
+  // 1. seleniumServerJar - to start Selenium Standalone locally.
+  // 2. seleniumAddress - to connect to a Selenium server which is already
+  //    running.
+  // 3. sauceUser/sauceKey - to use remote Selenium servers via SauceLabs.
+  //
+  // If the chromeOnly option is specified, no Selenium server will be started,
+  // and chromeDriver will be used directly (from the location specified in
+  // chromeDriver)
+
+  // The location of the selenium standalone server .jar file, relative
+  // to the location of this config. If no other method of starting selenium
+  // is found, this will default to
+  // node_modules/protractor/selenium/selenium-server...
+  seleniumServerJar: null,
+  // The port to start the selenium server on, or null if the server should
+  // find its own unused port.
+  seleniumPort: null,
+  // Chromedriver location is used to help the selenium standalone server
+  // find chromedriver. This will be passed to the selenium jar as
+  // the system property webdriver.chrome.driver. If null, selenium will
+  // attempt to find chromedriver using PATH.
+  chromeDriver: './selenium/chromedriver',
+  // If true, only chromedriver will be started, not a standalone selenium.
+  // Tests for browsers other than chrome will not run.
+  chromeOnly: false,
+  // Additional command line options to pass to selenium. For example,
+  // if you need to change the browser timeout, use
+  // seleniumArgs: ['-browserTimeout=60'],
+  seleniumArgs: [],
+  beforeLaunch: function() {
+    Object.keys(suites).forEach(function(key) {
+      var patterns = suites[key];
```

Comment:
Just curious, why "patterns"? Instead of, say, suite or filename or something like that?

Reply:
> why "patterns"?
suites[key] has a list of patterns. ex:
```
suites: {
    full: [
      'protractor/*.js',
      'protractor_desktop/*.js'
    ], ......
```

>  Instead of, say, suite

I think suite is the key (suites[key]) and the pattern is the general structure, not every time a pattern will be the exact file path. Should I change the key to the suite?